proxy/src/main/java/com/velocitypowered/proxy/scheduler/VelocityScheduler.java

```diff
@@ -65,12 +68,28 @@ public VelocityScheduler(PluginManager pluginManager) {
 
   @Override
   public TaskBuilder buildTask(Object plugin, Runnable runnable) {
-    checkNotNull(plugin, "plugin");
-    checkNotNull(runnable, "runnable");
+    requireNonNull(runnable, () -> "the scheduled task cannot be null");
+    requireNonNull(plugin, () -> "the plugin cannot be null");
     checkArgument(pluginManager.fromInstance(plugin).isPresent(), "plugin is not registered");
     return new TaskBuilderImpl(plugin, runnable);
   }
 
+  @Override
+  public TaskBuilder builder(Object plugin) {
+    requireNonNull(plugin, () -> "the plugin cannot be null");
+    checkArgument(pluginManager.fromInstance(plugin).isPresent(), "plugin is not registered");
+    return new TaskBuilderImpl(plugin, null);
+  }
+
+  @Override
+  public @NonNull Collection<ScheduledTask> tasksByPlugin(@NonNull Object plugin) {
+    requireNonNull(plugin, () -> "the plugin cannot be null");
+    checkArgument(pluginManager.fromInstance(plugin).isPresent(), "plugin is not registered");
+    synchronized (tasksByPlugin) {
+      return Set.copyOf(tasksByPlugin.get(plugin));
```

Comment:
Nit: the `get` call can be moved outside of the synchronized block (https://javadoc.io/doc/com.google.guava/guava/latest/com/google/common/collect/Multimaps.html#synchronizedMultimap(com.google.common.collect.Multimap) (the synchronized block would then contain `Set.copyOf(tasks)`.